lib/exportModules.nix

```diff
@@ -21,35 +30,20 @@ let
 
       **/
 
-    let
-
-      removeSuffix = suffix: str:
-        let
-          sufLen = builtins.stringLength suffix;
-          sLen = builtins.stringLength str;
-        in
-        if sufLen <= sLen && suffix == builtins.substring (sLen - sufLen) sufLen str then
-          builtins.substring 0 (sLen - sufLen) str
-        else
-          str;
-
-      genAttrs' = func: values: builtins.listToAttrs (map func values);
-
-      hasFileAttr = o: builtins.hasAttr "_file" o;
-      peek = f: f (builtins.functionArgs f);
-
-    in
-
     genAttrs'
       (arg:
 
-        # a regular path to be imported
-        if builtins.isPath arg then
+        # a regular file to be imported
+        if pathIsRegularFile arg then
           {
             name = removeSuffix ".nix" (baseNameOf arg);
             value = import arg;
           }
 
+        # a directory to be recursively imported
+        else if pathIsDirectory arg then
+          rakeLeaves arg
```

Comment:
Just noticed: since this is intended to be consumed by `self.nixosModules`, this also needs to implement `digga.flattenTree`. Otherwise this will produce illegal (nested) output. See `flake check`.

Reply:
unfortunately, this seems to cause a regression:
```console
error: 'name' attribute missing in a call to 'listToAttrs'

       at /nix/store/qqv2yvbj4ra9hpf8ar4bih9wc1fxwvs0-source/lib/internal-functions.nix:13:29:

           12|   */
           13|   genAttrs' = func: values: builtins.listToAttrs (map func values);
             |                             ^
           14|

       … while evaluating 'genAttrs''

       at /nix/store/qqv2yvbj4ra9hpf8ar4bih9wc1fxwvs0-source/lib/internal-functions.nix:13:21:

           12|   */
           13|   genAttrs' = func: values: builtins.listToAttrs (map func values);
             |                     ^
           14|

       … from call site

       at /nix/store/qqv2yvbj4ra9hpf8ar4bih9wc1fxwvs0-source/lib/exportModules.nix:33:5:

           32|
           33|     genAttrs'
             |     ^
           34|       (arg:

       … while evaluating 'exportModules'

       at /nix/store/qqv2yvbj4ra9hpf8ar4bih9wc1fxwvs0-source/lib/exportModules.nix:13:19:

           12|     ;
           13|   exportModules = args:
             |                   ^
           14|     /**

       … from call site

       at /nix/store/c2jawjpd85s319ql9n4fsnc88krbbjsj-source/src/mkFlake/fup-adapter.nix:88:23:

           87|
           88|     devshellModules = flake-utils-plus.lib.exportModules config.devshell.modules;
             |                       ^
           89|

       … while checking flake output 'devshellModules'

       at /nix/store/c2jawjpd85s319ql9n4fsnc88krbbjsj-source/src/mkFlake/fup-adapter.nix:88:5:

           87|
           88|     devshellModules = flake-utils-plus.lib.exportModules config.devshell.modules;
             |     ^
           89|
```

I'll be reverting. Maybe `rakeLeaves` is not the right tool, here.